meps/mep-cross-references.md

```diff
@@ -0,0 +1,339 @@
+- If the node is not enumerated and does not have a title, the reference label will be used.
+
+If the `text` is included it will be used as is with two additional template values (`%s` and `%t`)
+
+- Enumeration (`%s`)
+
+  - Any reference target that is enumerated can reference that number or string with a `%s`.
+  - If the target is enumerated, the default will be the numbered form of the reference (e.g. "Section 2.1.2", "Fig. 3", or "(1)")
+  - If a `%s` is used and the node is not enumerated, the `%s` will be replaced by "??" and a warning raised.
+  - Parsers can optionally choose to support `{number}` which is from Sphinx.
+
+- Title (`%t`)
+  - Any node can include the title of a reference including any styles (e.g. Sections are the section title; Figures and Tables are the caption).
+  - If a `%t` is used and the node does not have an explicit name or title, the node reference label will be used.
+  - Parsers can optionally choose to support `{name}` which is from Sphinx.
+
+#### `link`
+
+The links are defined by a scheme, which can be standard protocols (`http:`, `mailto:`). Here we propose three new schemes, `path`, `project`, and `myst`, which is an extensibility point described by [CommonMark](https://spec.commonmark.org/0.30/#example-598). These schemes are used to indicate that the link should be resolved by MyST specific logic, and follows standard [URI][uri] syntax:
+
+```text
+URI = scheme ":" pathname ["?" query] "#" fragment
+```
+
+In most cases, as seen in the summary above the scheme is not required to be explicit and can be inferred safely by the context.
+The exception is when _explicitly_ referring to an external MyST site, Jupyter Book or Sphinx documentation site.
+These URIs can be safely and easily parsed by any common URL parser. For example in Javascript:
+
+```js
+const url = new URL('project:target.md#my-ref');
+url.protocol; // "project:"
+url.pathname; // "target.md"
+url.hash; // "#my-ref"
+```
+
+For most internally linked references, we expect the inline syntax to be most commonly used, with the autolink and scheme to only be used when specific behavior is intended. The following links and references are supported:
+
+| Link Type                | Auto Link                  | Inline                    |
+| :----------------------- | :------------------------- | :------------------------ |
+| External URL             | `<https://example.com>`    | `[](https://example.com)` |
+| Local file download      | `<path:file.txt>`          | `[](file.txt)`            |
+| File download (explicit) | `<path:file.md>`           | `[](path:file.md)`        |
+| Project document         | `<project:file.md>`        | `[](file.md)`             |
+| Target in a document     | `<project:target.md#file>` | `[](file.md#target)`      |
+| Target in project        | `<project:#target>`        | `[](#target)`             |
+
+### Search Order and Specificity
```

Comment:
One thing that is not mentioned here is the "fall back" behaviour, i.e. what happens if a link cannot be resolved.

In myst-parser I recently changed this from outputting nothing to outputting an external link, i.e. `<a href="#target">#target</a>`

Reply:
Put in:

> If a link cannot be resolved, an external link should be rendered, for example, `<a href="#target">#target</a>`.